lib/lotus/model/adapters/sql/command.rb

```diff
@@ -55,10 +59,41 @@ def update(entity)
           # @api private
           # @since 0.1.0
           def delete
-            @collection.delete
+            _rescue_database_error do
+              @collection.delete
+            end
           end
 
           alias_method :clear, :delete
+
+          private
+
+          # Rescues Sequel::DatabaseError in yielded block
+          #
+          # @api private
+          def _rescue_database_error
+            yield
+          rescue Sequel::DatabaseError => e
+            raise _transform_database_error(e)
+          end
+
+          # Transforms error into a Lotus::Model::Error
+          #
+          # @api private
+          def _transform_database_error(error)
+            case error
+            when Sequel::CheckConstraintViolation
+              Lotus::Model::CheckConstraintViolationError.new(error.message)
+            when Sequel::ForeignKeyConstraintViolation
+              Lotus::Model::ForeignKeyConstraintViolationError.new(error.message)
+            when Sequel::NotNullConstraintViolation
+              Lotus::Model::NotNullConstraintViolationError.new(error.message)
+            when Sequel::UniqueConstraintViolation
+              Lotus::Model::UniqueConstraintViolationError.new(error.message)
+            else
+              Lotus::Model::InvalidCommandError.new(error.message)
```

Comment:
I was preserving the behavior in `SqlAdapter#execute` (https://github.com/lotus/model/blob/v0.5.1/lib/lotus/model/adapters/sql_adapter.rb#L244-L246), where all `Sequel::DatabaseError` were transformed into `Lotus::InvalidCommandError`.  However, given https://github.com/lotus/model/pull/281#discussion_r49831938, I'll just raise `Lotus::Model::Error` irregardless of the specific error subclass.